From be4213de25403d808f1c3e9c45030119091eff77 Mon Sep 17 00:00:00 2001
From: Florian Maurer <f.maurer@outlook.de>
Date: Thu, 21 Dec 2023 15:35:57 +0100
Subject: [PATCH 1/2] fritzbox-7530: utilize lan1 as wan, others as lan With
 the switch to DSA on this board, we can now utilize the ports better, by
 splitting them according to their usage

Signed-off-by: Florian Maurer <f.maurer@outlook.de>
---
 package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
index 9c3f1dc9b66..5e5243a0a15 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
@@ -63,6 +63,10 @@ elseif platform.match('ath79', 'generic', {
 	'ubnt,unifiac-pro',
 }) then
 	lan_ifname, wan_ifname = 'eth0.2', 'eth0.1'
+elseif platform.match('ipq40xx', 'generic', {
+	'avm,fritzbox-7530',
+}) then
+	lan_ifname, wan_ifname = 'lan2 lan3 lan4', 'lan1'
 elseif platform.match('ramips', 'mt7621', {
 	'netgear,wac104',
 }) then

From 27fe1ed4e28c31fdbc0c84366957ad4329324c03 Mon Sep 17 00:00:00 2001
From: Grische <github@grische.xyz>
Date: Tue, 9 Jul 2024 18:25:54 +0200
Subject: [PATCH 2/2] gluon-core: migrate lan role of single port config

when a device is migrated from a single WAN/LAN configuration to a
multi-port configuration (e.g. using DSA), set the same role on all
ports and not just the new WAN port.
---
 package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles | 1 +
 1 file changed, 1 insertion(+)

diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles b/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles
index 6bceb36e55b..3040341155f 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles
@@ -43,6 +43,7 @@ if uci:get('gluon', 'iface_wan') then
 end
 if uci:get('gluon', 'iface_single') then
 	roles.wan = uci:get_list('gluon', 'iface_single', 'role')
+	roles.lan = uci:get_list('gluon', 'iface_single', 'role')
 end
 
 -- Non-existing interfaces are nil, so they will not be added to the table
